larq_compute_engine/mlir/python/graphdef_tfl_flatbuffer.cc

```diff
@@ -95,25 +106,25 @@ pybind11::bytes ConvertGraphDefToTFLiteFlatBuffer(
           default_ranges.cast<std::pair<double, double>>();
     }
   }
-  mlir::PassManager pm(&context);
+
+  mlir::PassManager pm(&context, mlir::OpPassManager::Nesting::Implicit);
+  tensorflow::SetCrashReproducer(pm);
+
   tensorflow::AddTFToLCETFLConversionPasses(
       quant_specs, &pm, target, experimental_enable_bitpacked_activations);
 
   // Convert back to outlined while format for export back to flatbuffer.
   pm.addPass(mlir::TFL::CreateWhileOutlinePass());
   pm.addPass(mlir::TFL::CreateRuntimeVerifyPass());
 
-  if (failed(pm.run(*module.ValueOrDie()))) {
-    throw std::runtime_error("Could not complete conversion passes.");
-  }
-
-  TruncateOpOrArgLocNameMapper op_or_arg_name_mapper;
-  // Write MLIR TFLite dialect into FlatBuffer
   std::string result;
-  if (tflite::MlirToFlatBufferTranslateFunction(
-          *module.ValueOrDie(), &result, /*emit_builtin_tflite_ops=*/true,
-          /*emit_select_tf_ops=*/false, /*emit_custom_ops=*/true,
-          &op_or_arg_name_mapper)) {
+  auto status = ConvertTFExecutorToTFLOrFlatbuffer(
```

Comment:
I previously used the lower lever `MlirToFlatBufferTranslateFunction` API  so we do not need rely on `@org_tensorflow//tensorflow/compiler/mlir/lite:tf_to_tfl_flatbuffer`. Last time I checked this target included many for us unneeded dependencies which I I opted to not use it. However, I am not sure if this is relevant anymore since our builds take ages anyway.
Do we gain an advantage in using `ConvertTFExecutorToTFLOrFlatbuffer` over `MlirToFlatBufferTranslateFunction` here?

Reply:
Yeah I was getting a bunch of build errors from the invokation of `MlirToFlatBufferTranslateFunction` which is why I went 'up a level' here in order to get it working. Also I guess I thought mirroring what TF do more closely would make the upgrade easier, which indeed it did :p

I didn't realise though that the extra dependency would bring in a lot of transitive deps that weren't already having to build anyway - do you know which bits those are and why we're able to avoid them but TF aren't? For that matter, what does `ConvertTFExecutorToTFLOrFlatbuffer` actually do beyond what we need it to? To be clear I'm super happy to try and revert to the previous method, this is just for my understanding :)

Reply:
> do you know which bits those are and why we're able to avoid them but TF aren't

I can check if this is actually still the case in TF 2.5, but last time I checked it included some of the flatbuffer import logic that we don't need. I will look into this and see if this is still relevant since we are including most of TF already anyway.

Reply:
I checked using `bazelisk query "deps(:build_pip_pkg)" --notool_deps` and compared the diff. It looks like `tf_to_tfl_flatbuffer` adds to following targets:
```
@org_tensorflow//tensorflow/compiler/mlir/lite:common
@org_tensorflow//tensorflow/compiler/mlir/lite:common/tfl_pass_config.h
@org_tensorflow//tensorflow/compiler/mlir/lite:flatbuffer_import
@org_tensorflow//tensorflow/compiler/mlir/lite:flatbuffer_import.cc
@org_tensorflow//tensorflow/compiler/mlir/lite:flatbuffer_import.h
@org_tensorflow//tensorflow/compiler/mlir/lite:flatbuffer_translate_lib
@org_tensorflow//tensorflow/compiler/mlir/lite:tf_to_tfl_flatbuffer
@org_tensorflow//tensorflow/compiler/mlir/lite:tf_to_tfl_flatbuffer.cc
@org_tensorflow//tensorflow/compiler/mlir/lite:tf_to_tfl_flatbuffer.h
@org_tensorflow//tensorflow/compiler/mlir/tensorflow:tf_dialect_lib
@org_tensorflow//tensorflow/compiler/mlir/tensorflow:translate/tf_mlir_translate.cc
@org_tensorflow//tensorflow/compiler/mlir/tensorflow:translate/tf_mlir_translate.h
@org_tensorflow//tensorflow/compiler/mlir/tensorflow:translate_lib
@org_tensorflow//tensorflow/lite/tools/optimize:quantize_weights
@org_tensorflow//tensorflow/lite/tools/optimize:quantize_weights.cc
@org_tensorflow//tensorflow/lite/tools/optimize:quantize_weights.h
```
TBH this doesn't seem to be as bad as I remember. Maybe this doesn't matter anymore since we now include the fallback kernel implementations anyway.